### PR TITLE
Fixes #8687: Inconsistent directive API parameters at creation

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/8687

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

File: rudder-web/src/main/scala/com/normation/rudder/web/snippet/administration/PropertiesManagement.scala [hunk 1]
```diff
@@ -694,7 +681,50 @@ class PropertiesManagement extends DispatchSnippet with Loggable {
...
+
+    // form values
+    var apiMode  = initApiMode.getOrElse(true)
 
+    def submit = {
+      configService.set_api_compatibility_mode(apiMode).foreach(updateOk => initApiMode = Full(apiMode))
+
+      // start a promise generation, Since we check if there is change to save, if we got there it mean that we need to redeploy
+      S.notice("apiModeMsg", "Api compatibility mode changed")
+      check()
```
Comment (ncharles) [inline rudder-web/src/main/scala/com/normation/rudder/web/snippet/administration/PropertiesManagement.scala]: why should it start a generation ? it has only impact on the AP, not the generated promises

File: rudder-core/src/main/resources/Migration/ldapMigration-2.11-2.11-add-api-compatibility-mode.ldif [hunk 2]
```diff
@@ -0,0 +1,40 @@
+###############################################################################
+# Copyright 2014 Normation SAS
```
Comment (ncharles) [inline rudder-core/src/main/resources/Migration/ldapMigration-2.11-2.11-add-api-compatibility-mode.ldif]: 2016 ?

**Comment by jooooooon** [comment]
If there's a Migration/ldapMigration-2.11-2.11-add-api-compatibility-mode.ldif file, this needs another PR to actually call the migration!

**Comment by jooooooon** [comment]
Oh, I see, that's Normation/rudder-packages#995

**Comment by VinceMacBuche** [comment]
héhé

**Comment by jooooooon** [comment]
@VinceMacBuche Please merge when ready.

**Comment by VinceMacBuche** [comment]
OK, merging this PR

**Pull request merged**